### ISSUE TITLE
clarify the differences between defining a prefix and requiring the use of that pattern in a slot's pattern

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/nmdc-schema#978]
@aclum and @sujaypatil96 I thing we could link some additional issues here 

goals: 
- make sure we all understand the difference
- make sure that both aspects are handled well in the schema for prominent identifiers

----

- #965 
- #930 
- #819 
- #951

**Comment by aclum** [comment]
relates to https://github.com/microbiomedata/nmdc-schema/issues/913

**Comment by turbomam** [comment]
I just wrote a prefix summarizer!

**Comment by turbomam** [comment]
hmm, maybe that is missing some prefixes defined in imported modules...

**Comment by aclum** [comment]
This list is missing gnps

**Comment by turbomam** [comment]
What is the expansion for `gnps`?

**Comment by turbomam** [comment]
slot `gnps_task_identifiers` requires a `pattern` of "^gnps\\.task:[a-f0-9]+$"

I guess I need to do a better job of clarifying the difference between requiring a pattern and defining a prefix.

**Comment by turbomam** [comment]
Is this what you want to use?

https://bioregistry.io/registry/gnps.task

**Comment by aclum** [comment]
Yes, I thought for every prefix a corresponding expansion was provided by
the schema. Did gnps.task get left out accidentally or did I misunderstand
the discussion at the infrastructure meeting yesterday?

**Comment by turbomam** [comment]
It is better to have both a prefix definition and a pattern requirement. Either I forgot or clobbered some file I was working on. Thanks for checking.

I'll add this now:

```yaml
gnps.task: "https://bioregistry.io/reference/gnps.task:"
```

**Comment by aclum** [comment]
@turbomam  has added some great documentation about prefixes and how to find them to [here](https://github.com/microbiomedata/nmdc-schema/blob/main/src/docs/prefixes_curies_ids_mappings_etc.md).
The only thing that would be good to clarify is when to use slot `pattern` to enforce a prefix vs `structured_pattern.syntax`. Is `structured_pattern.syntax` only used in a local definition? ie when defined globally slot `id` uses 
`pattern: ^[a-zA-Z0-9][a-zA-Z0-9_\.]+:[a-zA-Z0-9_][a-zA-Z0-9_\-\/\.,]*$
`
but when id is defined locally for Class `Biosample` it uses 
```
structured_pattern:
      syntax: '{id_nmdc_prefix}:bsm-{id_shoulder}-{id_blade}{id_version}{id_locus}'
```
